services/search/src/search/routes/filter.py

```diff
@@ -0,0 +1,334 @@
+                    result = get_cache_entry_from_steps(
+                        processing_steps=[processing_steps],
+                        dataset=dataset,
+                        config=config,
+                        split=split,
+                        processing_graph=processing_graph,
+                        hf_endpoint=hf_endpoint,
+                        hf_token=hf_token,
+                        hf_timeout_seconds=hf_timeout_seconds,
+                        cache_max_days=cache_max_days,
+                    )
+                    content = result["content"]
+                    http_status = result["http_status"]
+                    error_code = result["error_code"]
+                    revision = result["dataset_git_revision"]
+                    if http_status != HTTPStatus.OK:
+                        return get_json_error_response(
+                            content=content,
+                            status_code=http_status,
+                            max_age=max_age_short,
+                            error_code=error_code,
+                            revision=revision,
+                        )
+                # TODO: duplicated in /search
+                with StepProfiler(method="search_endpoint", step="download index file if missing"):
+                    file_name = content["filename"]
+                    index_folder = get_download_folder(duckdb_index_file_directory, dataset, config, split, revision)
+                    # For directories like "partial-train" for the file
+                    # at "en/partial-train/0000.parquet" in the C4 dataset.
+                    # Note that "-" is forbidden for split names, so it doesn't create directory names collisions.
+                    split_directory = content["url"].rsplit("/", 2)[1]
+                    repo_file_location = f"{config}/{split_directory}/{file_name}"
+                    index_file_location = f"{index_folder}/{repo_file_location}"
+                    index_path = Path(index_file_location)
+                    if not index_path.is_file():
+                        with StepProfiler(method="search_endpoint", step="download index file"):
+                            download_index_file(
+                                cache_folder=f"{duckdb_index_file_directory}/{HUB_DOWNLOAD_CACHE_FOLDER}",
+                                index_folder=index_folder,
+                                target_revision=target_revision,
+                                dataset=dataset,
+                                repo_file_location=repo_file_location,
+                                hf_token=hf_token,
+                            )
+                with StepProfiler(method="filter_endpoint", step="get features"):
+                    if "features" in content and isinstance(content["features"], dict):
+                        features = Features.from_dict(content["features"])
```

Comment:
@huggingface/datasets-server can we safely assume that this condition is always true?

Reply:
i think `content["features"]` also might be `None` if i'm not mistaken. not sure though

Reply:
Thanks @Polina for your answer. :hugs: 
Any idea about in which cases can it be None?

Reply:
You can raise an error if it's None imo.

It can only happen in very rare cases (parquet dataset processed with old version of `datasets` for example)

Reply:
Thanks, I am removing the else branch as well and all the parquet metadata functions! :hugs: